quickjs.c

```diff
@@ -6360,72 +6386,141 @@ static void build_backtrace(JSContext *ctx, JSValue error_obj,
                             int backtrace_flags)
 {
     JSStackFrame *sf;
-    JSValue str;
+    JSValue stack, prepare, saved_exception;
     DynBuf dbuf;
     const char *func_name_str;
     const char *str1;
     JSObject *p;
-    BOOL backtrace_barrier;
-    int flags;
+    JSFunctionBytecode *b;
+    BOOL backtrace_barrier, has_prepare;
+    JSRuntime *rt;
+    JSCallSiteData *csd;
+    int csd_count;
+    uint32_t i;
 
-    flags = JS_PROP_WRITABLE | JS_PROP_CONFIGURABLE;
-    js_dbuf_init(ctx, &dbuf);
-    if (filename) {
-        dbuf_printf(&dbuf, "    at %s", filename);
-        if (line_num != -1)
-            dbuf_printf(&dbuf, ":%d:%d", line_num, col_num);
-        dbuf_putc(&dbuf, '\n');
-        if (backtrace_flags & JS_BACKTRACE_FLAG_SINGLE_LEVEL)
+    rt = ctx->rt;
+    has_prepare = FALSE;
+    i = 0;
+    csd_count = 10;
```

Comment:
@bnoordhuis What is the default in V8? What if there are more than 10, does it show that call depth, or does it stop there?

I see that `Error.stackTraceLimit` allows to customize that, which I was planning to add later, but I was wondering if I could `alloca` this thing perhaps to save the double `malloc` in the common / reasonable case?

Reply:
Yes, it stops there. I'd say that's desirable because unbounded stack depths can get expensive, CPU and memory wise.

If you put in some reasonable hard upper limit, like 64 or 128, you can keep it as an on-stack array, no alloca needed.

Reply:
Excellent! I'll try to wrap it up tomorrow then.